### PR TITLE
docs: pull who uses Renovate image into main repo

### LINKED ISSUE #28924: Move the Renovate matrix image into this repository
### Describe the proposed change(s).

Move the Renovate Matrix image into the `renovatebot/renovate` repository:

- [x] Copy/paste the source image into the `docs/usage/assets/images` folder, keeping the name
- [x] Update links to `https://app.renovatebot.com/images/matrix.png` so they point to the new location:
   - [x] Homepage file
   - [x] Readme file
   - [x] Docs files

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Pull the "Who uses Renovate" image (`matrix.png`) into the main repository

## Context

`@rarkins` wants to deprecate the `https://app.renovatebot.com/` domain. The `matrix.png` image is _one_ of the images we host over there.

We still host two images on that domain after this PR:

- A Renovate banner image _with_ Mend branding (used on docs homepage)
- A Renovate banner image with _no_ Mend branding (used in this repository's readme)

I'll pull in the banner images with _another_ PR, once I'm clear on how to replace them. I already asked on Slack what to do. :smile:  

Closes #28924.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/index.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ Renovate works on these platforms:
 
 Renovate is used by:
 
-![Renovate Matrix](https://app.renovatebot.com/images/matrix.png){ loading=lazy }
+![Renovate Matrix](../usage/assets/images/matrix.png){ loading=lazy }
```
Comment (HonkingGoose) [inline docs/usage/index.md:79]: As usual, please double check me on the links! :melting_face:

File: readme.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ Renovate works on these platforms:
 
 Renovate is widely used in the developer community:
 
-![Renovate Matrix](https://app.renovatebot.com/images/matrix.png)
+![Renovate Matrix](../renovate/docs/usage/assets/images/matrix.png)
```
Comment (HonkingGoose) [inline readme.md:48]: Same link check request here. :melting_face:

File: docs/usage/index.md (hunk 1 above)
Comment (viceice) [reply docs/usage/index.md:79]:
```suggestion
![Renovate Matrix](./assets/images/matrix.png){ loading=lazy }
```

File: readme.md (hunk 2 above)
Comment (viceice) [reply readme.md:48]: see above

**Review by viceice — COMMENTED** [review]
see comments

**Review by viceice — APPROVED** [review]
🤞

**Review by viceice — APPROVED** [review]

**Pull request merged**